packages/cli/src/eventbus/audit-event-relay.service.ts

```diff
@@ -121,10 +121,37 @@ export class AuditEventRelay {
 		});
 	}
 
-	private workflowPostExecute(event: Event['workflow-post-execute']) {
+	private workflowPostExecute({
+		executionId,
+		success,
+		userId,
+		workflowId,
+		isManual,
+		workflowName,
+		metadata,
+		lastNodeExecuted,
+		errorNodeType,
+		errorMessage,
+	}: Event['workflow-post-execute']) {
+		const payload: Event['workflow-post-execute'] = {
```

Comment:
We should have proper types for the `n8n.workflow.success` and `n8n.workflow.failed` events. They are not the same events as `Event['workflow-post-execute']` which `EventRelay` is emitting. Also their contents are different

Reply:
So there's a difference between the emitted payload type and the payload type used by the event bus. Let's improve types for the event bus later on once we finish tearing down internal hooks?

Reply:
Yes. We can think of the events emitted to the `EventService` something similar to Domain Events. Then we have different parties (e.g. AuditEventRelay, Telemetry) who are interested in these events and react to them by for example sending an audit event, or sending a telemetry event.